### PR TITLE
adding confidence scoring option for Kraken

### PR DESCRIPTION
**Pull request by meenachakra**
modified config and Snakefile

### CODE REVIEW

File: Snakefile [hunk 1]
```diff
@@ -79,6 +79,9 @@ else:
     paired_string = ''
 sample_names = sample_reads.keys()
 
+# also read in desired confidence threshold for Kraken
+confidence_threshold = config['confidence_threshold']
```
Comment (bsiranosian) [inline Snakefile:85]: To stop this from breaking previous config files, add something like 
```
if not 'confidence_threshold' in config: 
    config['confidence_threshold'] = 0.0
```

File: Snakefile [hunk 2]
```diff
@@ -176,7 +179,8 @@ rule kraken:
         krak_report = join(outdir, "classification/{samp}.krak.report")
     params:
         db = config['database'],
-        paired_string = paired_string
+        paired_string = paired_string,
+	confidence_threshold = confidence_threshold
```
Comment (bsiranosian) [inline Snakefile:185]: fix indentation level

File: Snakefile [hunk 3]
```diff
@@ -185,7 +189,8 @@ rule kraken:
     # singularity: "docker://quay.io/biocontainers/kraken2:2.0.9beta--pl526hc9558a2_0"
     shell: """
         time kraken2 --db {params.db} --threads {threads} --output {output.krak} \
-        --report {output.krak_report} {params.paired_string} {input.reads} --use-names
+        --report {output.krak_report} {params.paired_string} {input.reads} \
+	--confidence {params.confidence_threshold} --use-names
```
Comment (bsiranosian) [inline Snakefile:195]: Confidence = 0 functions as normal kraken, correct?

**Review by bsiranosian — CHANGES_REQUESTED** [review]

File: Snakefile (hunk 3 above)
Comment (meenachakra) [reply Snakefile:195]: Yes!

File: Snakefile (hunk 1 above)
Comment (meenachakra) [reply Snakefile:85]: Got it, good plan!

**Review by bsiranosian — APPROVED** [review]

**Pull request merged**